src/UnlockEligibility.sol

```diff
@@ -2,253 +2,256 @@
-   * 40      | hatId             | uint256     | 32      | HatsModule          |
-   * 72      | UNLOCK            | address     | 20      | {this}              |
-   * --------------------------------------------------------------------------+
-   */
-
-  /// @dev The configured Unlock factory contract address. If empty, the factory used will be determined by the mapping
-  /// defined within {getUnlockContract}.
-  function _UNLOCK() internal pure returns (address) {
-    return (_getArgAddress(72));
-  }
-
-  /*//////////////////////////////////////////////////////////////
+    /// @notice The default version of the lock contract deploy along with an instance of this module
+    uint16 public constant DEFAULT_LOCK_VERSION = 14;
+
+    /// @notice The address to split key purchase fees to
+    address public immutable FEE_SPLIT_RECIPIENT;
+
+    /// @notice The percentage of key purchase fees to split, in basis points (10000 = 100%)
+    uint256 public immutable FEE_SPLIT_PERCENTAGE;
+
+    /**
+     * This contract is a clone with immutable args, which means that it is deployed with a set of
+     * immutable storage variables (ie constants). Accessing these constants is cheaper than accessing
+     * regular storage variables (such as those set on initialization of a typical EIP-1167 clone),
+     * but requires a slightly different approach since they are read from calldata instead of storage.
+     *
+     * Below is a table of constants and their location.
+     *
+     * For more, see here: https://github.com/Saw-mon-and-Natalie/clones-with-immutable-args
+     *
+     * --------------------------------------------------------------------------+
+     * CLONE IMMUTABLE "STORAGE"                                                 |
+     * --------------------------------------------------------------------------|
+     * Offset  | Constant          | Type        | Length  | Source              |
+     * --------------------------------------------------------------------------|
+     * 0       | IMPLEMENTATION    | address     | 20      | HatsModule          |
+     * 20      | HATS              | address     | 20      | HatsModule          |
+     * 40      | hatId             | uint256     | 32      | HatsModule          |
+     * 72      | UNLOCK            | address     | 20      | {this}              |
+     * --------------------------------------------------------------------------+
+     */
+
+    /// @dev The configured Unlock factory contract address. If empty, the factory used will be determined by the mapping
+    /// defined within {getUnlockContract}.
+    function _UNLOCK() internal pure returns (address) {
+        return (_getArgAddress(72));
+    }
+
+    /*//////////////////////////////////////////////////////////////
                             MUTABLE STATE
   //////////////////////////////////////////////////////////////*/
 
-  IPublicLock public lock;
-
-  uint256 public keyPrice;
```

Comment:
Removed that one because I don't think we need it. The hook can in fact query the lock to get its current price. this would letthe lock manager change the price of the memberships

Reply:
yeah, I know re-entry has a bit of a gas cost, but I think in this case, it is safer.